### PR TITLE
[`flake8-bugbear`] Treat `raise NotImplemented`-only bodies as stub functions

### PR DESCRIPTION
**Pull request by Philipp-Thiel**
## Summary

As discussed in https://github.com/astral-sh/ruff/issues/10083#issuecomment-1969653610, stubs detection now also covers the case where the function body raises NotImplementedError and does nothing else.

## Test Plan

Tests for the relevant cases were added in B006_8.py

### CODE REVIEW

**Review by charliermarsh — APPROVED** [review]
Thanks!

File: crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs [hunk 1]
```diff
@@ -223,6 +227,15 @@ fn is_stub(function_def: &ast::StmtFunctionDef) -> bool {
...
             )
         }
+        Stmt::Raise(StmtRaise {
+            range: _,
+            exc: exception,
+            cause: _,
+        }) => exception.as_ref().is_some_and(|exc| {
+            semantic
+                .resolve_builtin_symbol(map_callable(exc))
+                .is_some_and(|name| matches!(name, "NotImplementedError" | "NotImplemented"))
```
Comment (charliermarsh) [inline crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs:237]: Changed this to use the semantic model, and allow `raise NotImplementedError` (without argument) and `raise NotImplemented`.

Comment (NeilGirdhar) [reply crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs:237]: Why do you allow `raise NotImplemented`?  `NotImplemented` is not an exception.  It should never be raised.  `NotImplemented` can be returned by comparison operators.

File: crates/ruff_linter/resources/test/fixtures/flake8_bugbear/B006_8.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    """This one raises a different exception"""
+    raise IndexError()
+
+
+def qux(a: list = []):
+    raise NotImplementedError
+
+
+def quux(a: list = []):
+    raise NotImplemented
```
Comment (NeilGirdhar) [inline crates/ruff_linter/resources/test/fixtures/flake8_bugbear/B006_8.py:20]: What's the purpose of this example?

File: crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs (hunk 1 above)
Comment (charliermarsh) [reply crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs:237]: We have separate rules to guard against `raise NotImplemented`. But if the entire function is `raise NotImplemented`, it's almost certainly intended to be `raise NotImplementedError`, right?

File: crates/ruff_linter/resources/test/fixtures/flake8_bugbear/B006_8.py (hunk 2 above)
Comment (charliermarsh) [reply crates/ruff_linter/resources/test/fixtures/flake8_bugbear/B006_8.py:20]: To verify that we treat `raise NotImplemented` like `raise NotImplementedError`. (Just to be clear, these aren't examples, they're tests.)

File: crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs (hunk 1 above)
Comment (NeilGirdhar) [reply crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs:237]: I see, fair enough!

**Pull request merged**